NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,568 @@
+
+import NuRadioReco.framework.event
+import NuRadioReco.framework.station
+import NuRadioReco.framework.channel
+import NuRadioReco.framework.trigger
+
+from NuRadioReco.utilities import units
+import mattak.Dataset
+
+try:
+   from rnog_data.runtable import RunTable
+   imported_runtable = True
+except ImportError:
+   print("Import of run table failed. You will not be able to select runs! \n" 
+         "You can get the interface from GitHub: git@github.com:RNO-G/rnog-data-analysis-and-issues.git")
+   imported_runtable = False
+
+
+def baseline_correction(wfs, n_bins=128, func=np.median):
+   """
+   Simple baseline correction function. Determines baseline in discrete chuncks of "n_bins" with
+   the function specified (i.e., mean or median).
+   
+   Parameters
+   ----------
+   
+   wfs: np.array(n_events, n_channels, n_samples)
+      Waveforms of several events/channels.
+      
+   n_bins: int
+      Number of samples/bins in one "chunck". If None, calculate median/mean over entire trace. (Default: 128)
+      
+   func: np.mean or np.median
+      Function to calculate pedestal
+   
+   Returns
+   -------
+   
+   wfs_corrected: np.array(n_events, n_channels, n_samples)
+      Baseline/pedestal corrected waveforms
+   """
+    
+   # Example: Get baselines in chunks of 128 bins
+   # wfs in (n_events, n_channels, 2048)
+   # np.split -> (16, n_events, n_channels, 128) each waveform split in 16 chuncks
+   # func -> (16, n_events, n_channels) pedestal for each chunck
+   if n_bins is not None:
+      baseline_values = func(np.split(wfs, 2048 // n_bins, axis=-1), axis=-1)
+
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, n_bins % 2048, axis=0)
+   else:
+      baseline_values = func(wfs, axis=-1)
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, 2048, axis=0)
+          
+   # np.moveaxis -> (n_events, n_channels, 2048)
+   baseline_traces = np.moveaxis(baseline_traces, 0, -1)
+    
+   return wfs - baseline_traces
+
+
+def get_time_offset(trigger_type):
+   """ 
+   Mapping the offset between trace start time and trigger time (~ signal time). 
+   Temporary use hard-coded values for each trigger type. In the future this
+   information might be time, station, and channel dependent and should come 
+   from a database (or is already calibrated in mattak)
+   
+   Parameters
+   ----------
+   
+   trigger_type: str
+      Trigger type encoded as string from Mattak
+      
+   Returns
+   -------
+   
+   time_offset: float
+      trace_start_time = trigger_time - time_offset 
+   
+   """
+   
+   time_offsets = {
+      "FORCE": 0,
+      "LT": 213 * units.ns,  # ~ 1 / 3 of trace @ 2048 sample with 3.2 GSa/s
+      "RADIANT": 320 * units.ns  # ~ 1 / 2 of trace @ 2048 sample with 3.2 GSa/s
```

Comment:
do we want to set accurate timings here as in the field or not?

Reply:
sure, do we know them already? Those values are somewhat placeholders

Reply:
Any suggestions?

Reply:
for better place holders, maybe these plots are helpful:

Reply:
something of order 250ns for LT triggered, 450ns for surface